File: application/app/src/main/java/ir/ritmeapp/ritme/domain/model/PregnancyIntention.kt
package ir.ritmeapp.ritme.domain.model

/**
 * How the user relates to pregnancy, captured once at onboarding. This is Ritme's mode switch:
 * [PREGNANT] puts the account in pregnancy mode (cycle questions are skipped), everything else
 * is cycle mode. [apiValue] is the exact string the backend's `pregnancy_intention` enum expects
 * (`avoiding` / `pregnant` / `trying` / `unsure`); the server derives `user_goal` from it.
 */
enum class PregnancyIntention(val apiValue: String) {
    AVOIDING("avoiding"),
    PREGNANT("pregnant"),
    TRYING("trying"),
    UNSURE("unsure"),
    ;

    /** Cycle mode collects period/cycle data; pregnancy mode does not. */
    val isCycleMode: Boolean get() = this != PREGNANT

    companion object {
        /** Maps a backend token back to the enum; null for absent/unknown values. */
        fun fromApi(value: String?): PregnancyIntention? = entries.firstOrNull { it.apiValue == value }
    }
}

File: backend/app/Enums/PregnancyIntention.php
<?php

namespace App\Enums;

/**
 * How the user relates to pregnancy at onboarding time. Captured for women-only
 * users to branch the signup flow (pregnant → pregnancy mode, others → cycle
 * questions) and to seed the cycle engine's user_goal.
 */
enum PregnancyIntention: string
{
    case AVOIDING = 'avoiding';   // فعلاً قصد بارداری ندارم
    case PREGNANT = 'pregnant';   // باردار هستم
    case TRYING = 'trying';       // در حال تلاش برای بارداری
    case UNSURE = 'unsure';       // مطمئن نیستم / هنوز تصمیم نگرفته‌ام

    public static function values(): array
    {
        return array_column(self::cases(), 'value');
    }

    public function label(string $locale = 'en'): string
    {
        return match ($locale) {
            'fa' => match ($this) {
                self::AVOIDING => 'فعلاً قصد بارداری ندارم',
                self::PREGNANT => 'باردار هستم',
                self::TRYING => 'در حال تلاش برای بارداری هستم',
                self::UNSURE => 'مطمئن نیستم / هنوز تصمیم نگرفته‌ام',
            },
            default => match ($this) {
                self::AVOIDING => 'Not planning a pregnancy right now',
                self::PREGNANT => 'I am pregnant',
                self::TRYING => 'Trying to conceive',
                self::UNSURE => 'Not sure yet',
            },
        };
    }
}
